Source/Private/ASLayoutTransition.mm

```diff
@@ -98,27 +105,44 @@ - (BOOL)isSynchronous
 
 - (void)commitTransition
 {
-  [self applySubnodeInsertions];
   [self applySubnodeRemovals];
+  [self applySubnodeInsertionsAndMoves];
 }
 
-- (void)applySubnodeInsertions
+- (void)applySubnodeInsertionsAndMoves
 {
   ASDN::MutexSharedLocker l(__instanceLock__);
   [self calculateSubnodeOperationsIfNeeded];
   
   // Create an activity even if no subnodes affected.
-  as_activity_create_for_scope("Apply subnode insertions");
-  if (_insertedSubnodes.count == 0) {
+  as_activity_create_for_scope("Apply subnode insertions and moves");
+  if (_insertedSubnodePositions.size() == 0 && _subnodeMoves.size() == 0) {
     return;
   }
 
   ASDisplayNodeLogEvent(_node, @"insertSubnodes: %@", _insertedSubnodes);
   NSUInteger i = 0;
-  for (ASDisplayNode *node in _insertedSubnodes) {
+  NSUInteger j = 0;
+  for (j = 0; j < _subnodeMoves.size(); ++j) {
+    [_subnodeMoves[j].first _removeFromSupernodeIfEqualTo:_node];
```

Comment:
I think we can do `for (auto const &move : _subnodeMoves) { [move.first _remove…] }` and use C++ enumeration below (during the insertions) as well.

Reply:
I don't think it is so straightforward below (insertions/moves), as we are interleaving the two vectors and consuming from only one or the other each time.